src/messages/MClientReply.h

```diff
@@ -390,13 +388,23 @@ class MClientReply final : public MMDSOp {
     using ceph::decode;
     auto p = payload.cbegin();
     decode(head, p);
+    // for the backward compatability, assuming LINUX error codes on the wire,
+    // errorcode32_t implements conversion from/to different host error codes
+    if (header.version >= HEAD_VERSION) {
```

Comment:
If setters/getters for the result code are present, there is no need to do modify `head.result` during decode. Just transform in the getter/setter.

Reply:
i.e. keep `head.result` as the on-the-wire representation.

Reply:
This was the first idea, however, since head is protected, child of this class, if implemented, can access head.result directly. I prefer to have state of the object less fragile, in general. But if this approach is OK in general for the team, I don't mind.